mscore/pianoroll/pianoview.cpp

```diff
@@ -1245,4 +1302,155 @@ void PianoView::setSubdiv(int value)
             }
       }
 
+//---------------------------------------------------------
+//   cutNotes
+//---------------------------------------------------------
+
+void PianoView::cutNotes()
+      {
+      copyNotes();
+
+      Score* score = _staff->score();
+      score->startCmd();
+
+      score->cmdDeleteSelection();
+
+      score->endCmd();
+      }
+
+//---------------------------------------------------------
+//   copyNotes
+//---------------------------------------------------------
+
+void PianoView::copyNotes()
+      {
+      //Custom note copy routine based on Selection::staffMimeData()
+      Fraction firstTick;
+      bool init = false;
+      for (int i = 0; i < _noteList.size(); ++i) {
+            if (_noteList[i]->note()->selected()) {
```

Comment:
It is better to use the `at` method

Reply:
Why is that better than using operator[]?

Reply:
For two reasons: 
1. The `operator[]` returns a reference to the element by which the element can be changed. Therefore, the question arises - where and why the element is changing. The `at` method returns a constant reference, it is immediately clear that the element will not be changed. 
Therefore, if there is no need to change the element, it is better to use `at`, and if there is an intention to change the element, then the `operator[]`

2. Many C++ classes in Qt use implicit data sharing to maximize resource usage and minimize copying. Implicitly shared classes are both safe and efficient when passed as arguments, because only a pointer to the data is passed around, and the data is copied only if and when a function writes to it, i.e., copy-on-write.
(https://doc.qt.io/qt-5/implicit-sharing.html)  

Not for this case, but in for in the context of using the `operator[]` and the `at` method.
If the container is passed by the method argument or returned from the method by value, then there is no copy of the data until we try to change this data.
If you use an `operator[]`, this is a data change and a copy of the data will be made at the time the operator is called. If you use the `at` method, then there is no change in the data and there will be no copy.